2015/9-easy-exceptions-and-logging.pod

```diff
@@ -0,0 +1,293 @@
+message, then at some point it has to be translated into its L<Log::Report>
+equivalent. There is a bit of guesswork that goes into this (see the
+L<Log::Report> documentation), and sometimes it might not quite be what
+you want. For example, if the code in the above example C<die>s due to
+a coding error such as calling a method on an undefined value, then
+it will be reported as an C<ERROR>, rather than C<PANIC>. The former
+will display the message to the user, which is not what we want. I'm
+considering changing the code to report all such errors at the C<PANIC> level, as they
+are normally always unexpected application exceptions.
+
+
+=head2 Dispatchers
+
+I have purposefully tried to keep this article simple, without going into
+too much of the terminology of L<Log::Report>. There is one thing I should
+mention though, and that is the concept of dispatchers. A dispatcher is
+something that processes a message, and can be thought of a bit like
+a Dancer logging engine. You can have as many dispatchers as you want,
+and there are several types available (C<Log::Report::Dispatcher::*>). By
+default, a C<PERL> dispatcher is created for sending messages to STDERR,
+and a special C<Callback> dispatcher is used for sending the messages to the template.
+Dispatchers are added using the L<Dancer2::Logger::LogReport> configuration.
+
+The configuration of each dispatcher defines what messages it outputs. You
+might want one dispatcher to show all messages, but another to only display
+messages above a certain threshold. The C<mode> of a dispatcher defines what
+messages are shown. The default C<mode> is C<NORMAL>, which only shows messages
+at the level of C<NOTICE> and above. You can change the mode to C<VERBOSE>,
+C<ASSERT> or C<DEBUG> to get an increasing level of messages. If you don't
+specifiy a mode for a dispatcher, it will take the "global" value. The default
+global value is C<NORMAL>. The global value can be changed when loading the
+module:
+
+ use Dancer2::Plugin::LogReport mode => 'DEBUG';
+
+If you want finer control on what messages a dispatcher processes, you can
+also configure the C<accept> parameter. A full dispatcher configuration
+might look like this:
+
+ engines:
+   logger:
+     LogReport:
+       dispatchers:
+         default:              # Name. This one overrides the default dispatcher
+           type: PERL
+           mode: NORMAL
+           accept: FAILURE-    # Only FAILURE messages and above
+         syslog                # Additional dispatcher called syslog
+           type: SYSLOG
+           identity: myapp
+           facility: local0
+           flags: "pid ndelay nowait"
+           mode: DEBUG
+
+Dispatchers can also be used to format messages.
+
+=head2 There is a lot more
+
+The above is just a flavour of the common and easy stuff you might like
+to do. C<Log::Report> itself is way, way more powerful than that though.
+I'll hopefully write a follow-up article in the future on some of the
```

Comment:
"Some of the things that can be covered in a future article..."

Generally I would not say this, because then a person wonders whether that article exists. That's something to be determined before publishing it, obviously. I would instead say "There are even more advanced topic, such as...".

If you write the articles (yes, please) and we decide to publish them, you could add "Stay tuned for additional articles on these topics during this Advent Calendar" or something like that.